### PR TITLE
Standardise fit signature for ML and Bayes, fixes #620

### LINKED ISSUE #620: Make initial value passing consistent between ML and Bayesian fit methods
We currently pass the initial values as the third positional argument to the `fit` methods for ML while it's an optional fourth argument in the Bayesian `fit` method. It should be made consistent. The simpler solution is to deprecate the current Bayesian `fit` signature and introduce a new one similar to the ML `fit` methods.

This is somewhat related to #424 since the current Bayesian `fit` method relies on initial values from the domains.

### PR DESCRIPTION
**Pull request by Vaibhavdixit02**

### CODE REVIEW

File: src/estimation/bayes.jl [hunk 1]
```diff
@@ -130,7 +130,7 @@ struct BayesMCMC <: LikelihoodApproximation end
 #   BayesMCMCResults(bayes, chain, tuned)
 # end
 
-function Distributions.fit(model::PumasModel, data::Population, ::BayesMCMC, θ_init=init_param(model),
+function fit(model::PumasModel, data::Population, ::BayesMCMC, θ_init=init_param(model),
```
Comment (ChrisRackauckas) [inline src/estimation/bayes.jl]: Distributions.fit(model,data,param::BayesMCMC ...)

**Comment by andreasnoack** [comment]
Would be nice with a deprecation of the old signature.

File: src/estimation/bayes.jl [hunk 2]
```diff
@@ -157,6 +157,11 @@ function Distributions.fit(model::PumasModel, data::Population, ::BayesMCMC, θ_
   BayesMCMCResults(bayes, samples_, stats)
 end
 
+function Distributions.fit(model::PumasModel, data::Population, param::NamedTuple, ::BayesMCMC,
```
Comment (ChrisRackauckas) [inline src/estimation/bayes.jl]: it would be nice to include a dispatch without `param`

Comment (andreasnoack) [reply src/estimation/bayes.jl]: Using `init_param`?

Comment (Vaibhavdixit02) [reply src/estimation/bayes.jl]: @ChrisRackauckas if you look at the arguments of `fit` defined in Line 133, it allows calling `fit` without passing the parameters. So, maybe we shouldn't deprecate that?

Comment (ChrisRackauckas) [reply src/estimation/bayes.jl]: yeah that would work.

**Pull request merged**